site/src/modules/dashboard/Navbar/DeploymentDropdown.tsx

```diff
@@ -0,0 +1,137 @@
+import { css, type Interpolation, type Theme, useTheme } from "@emotion/react";
+import Button from "@mui/material/Button";
+import MenuItem from "@mui/material/MenuItem";
+import { type FC } from "react";
+import { NavLink } from "react-router-dom";
+import { DropdownArrow } from "components/DropdownArrow/DropdownArrow";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  usePopover,
+} from "components/Popover/Popover";
+
+import { USERS_LINK } from "modules/navigation";
+
+interface DeploymentDropdownProps {
+  canViewAuditLog: boolean;
+  canViewDeployment: boolean;
+  canViewAllUsers: boolean;
+  canViewHealth: boolean;
+}
+
+export const DeploymentDropdown: FC<DeploymentDropdownProps> = ({
+  canViewAuditLog,
+  canViewDeployment,
+  canViewAllUsers,
+  canViewHealth,
+}) => {
+  const theme = useTheme();
+
+  if (
+    !canViewAuditLog &&
+    !canViewDeployment &&
+    !canViewAllUsers &&
+    !canViewHealth
+  ) {
+    return null;
+  }
+
+  return (
+    <Popover>
+      <PopoverTrigger>
+        <Button>
+          Deployment
+          <DropdownArrow
+            color={theme.experimental.l2.fill.solid}
+            close={false}
+          />
+        </Button>
+      </PopoverTrigger>
+
+      <PopoverContent
+        horizontal="right"
+        css={{
+          ".MuiPaper-root": {
+            minWidth: "auto",
+            width: 180,
+            boxShadow: theme.shadows[6],
+          },
+        }}
+      >
+        <DeploymentDropdownContent
+          canViewAuditLog={canViewAuditLog}
+          canViewDeployment={canViewDeployment}
+          canViewAllUsers={canViewAllUsers}
+          canViewHealth={canViewHealth}
+        />
+      </PopoverContent>
+    </Popover>
+  );
+};
+
+const DeploymentDropdownContent: FC<DeploymentDropdownProps> = ({
+  canViewAuditLog,
+  canViewDeployment,
+  canViewAllUsers,
+  canViewHealth,
+}) => {
+  const popover = usePopover();
+
+  const onPopoverClose = () => popover.setIsOpen(false);
+
+  return (
+    <nav>
+      {canViewDeployment && (
+        <NavLink css={styles.link} to="/deployment/general">
+          <MenuItem css={styles.menuItem} onClick={onPopoverClose}>
+            Settings
+          </MenuItem>
+        </NavLink>
+      )}
+      {canViewAllUsers && (
+        <NavLink css={styles.link} to={USERS_LINK}>
+          <MenuItem css={styles.menuItem} onClick={onPopoverClose}>
+            Users
+          </MenuItem>
+        </NavLink>
+      )}
+      {canViewAuditLog && (
+        <NavLink css={styles.link} to="/audit">
+          <MenuItem css={styles.menuItem} onClick={onPopoverClose}>
```

Comment:
Instead of using `NavLink` as a wrapper, I think you can use the `component` prop in the `MenuLink` to simplify the code. 

```tsx
<MenuItem component={NavLink} to="/audit" onClick={...} />
```